File: Model/ViewModel/OrderDetailVM.cs
using targeteo.pl.Model.Entities;

namespace targeteo.pl.Model.ViewModel
{
    public class OrderDetailVM
    {
        public int OrderDetailId { get; set; }
        public string ForeignOrderId { get; set; }
        public string ProductId { get; set; }
        public int Quantity { get; set; }
        public double UnitPrice { get; set; }
        public ProductVM Product { get; set; }
        public OneUserOrderVM OneUserOrder { get; set; }
    }
}

File: ClientApp/src/app/model/orderDetailVm.ts
import { OneUserOrderVm } from "./oneUserOrderVm";
import { OrderVm } from "./orderVm";
import { ProductVm } from "./productVm";

export class orderDetailVm {
    
    public OrderDetailId: number;
    public ForeignOrderId: string;
    public ProductId: string;
    public Quantity: number;
    public UnitPrice: number;
    public Product: ProductVm;
    public OneUserOrder: OneUserOrderVm;
}